src/components/AttachmentModal.js

```diff
@@ -155,6 +155,9 @@ class AttachmentModal extends PureComponent {
      */
     closeConfirmModal() {
         this.setState({isAttachmentInvalid: false});
+        if (this.props.onModalHide) {
+            this.props.onModalHide();
+        }
```

Comment:
Should we expose the onConfirm/onCancel for the ConfirmModal as props to the AttachmentModal? The AttachmentModal is responsible for displaying the ConfirmModal, and it's only ever displayed if the attachment is invalid.

Reply:
> What if we pass a prop called onModalShow to AttachmentModal and call it on the innerModal's onModalShow. Now set the `isAttachmentPreviewActive` in this callback from `ReportActionCompose`

This sounds like it could work. I'll test it out.

Reply:
@rushatgabhane I've implemented @parasharrajat and it works great. Pushed up a new commit for review.